cram_pr2/cram_pr2_projection/src/tf.lisp

```diff
@@ -43,43 +44,81 @@
-     :suppress-callbacks t)
+      ;; tell the tf transformer that the global fixed frame and the odom frame are the same
+      (cl-tf:set-transform
+       transformer
+       (cl-tf:make-transform-stamped
+        fixed-frame odom-frame time
+        (cl-transforms:make-identity-vector)
+        (cl-transforms:make-identity-rotation))
+       :suppress-callbacks t)
 
-    ;; tell the tf transformer where the robot is in the odometry frame
-    (cl-tf:set-transform
-     transformer
-     (cl-tf:transform->transform-stamped
-      odom-frame base-frame time
-      (cl-transforms:pose->transform robot-pose-in-map))
-     :suppress-callbacks t)
+      ;; tell the tf transformer where the robot is in the odometry frame
+      (cl-tf:set-transform
+       transformer
+       (cl-tf:transform->transform-stamped
+        odom-frame base-frame time
+        (cl-transforms:pose->transform robot-pose-in-map))
+       :suppress-callbacks t)
 
-    ;; tell the tf transformer the current configuration of robot's joints
-    (dolist (link (btr:link-names ?robot-instance))
-      (unless (equal link base-frame)
-        (let ((transform (cl-transforms:transform*
-                          reference-transform-inv
-                          (cl-transforms:reference-transform
-                           (btr:link-pose ?robot-instance link)))))
-          (cl-tf:set-transform
-           transformer
-           (cl-tf:make-transform-stamped
-            base-frame link time
-            (cl-transforms:translation transform)
-            (cl-transforms:rotation transform))
-           :suppress-callbacks t))))
+      ;; tell the tf transformer the current configuration of robot's joints
+      (dolist (link (btr:link-names ?robot-instance))
+        (unless (equal link base-frame)
+          (let ((transform (cl-transforms:transform*
+                            reference-transform-inv
+                            (cl-transforms:reference-transform
+                             (btr:link-pose ?robot-instance link)))))
+            (cl-tf:set-transform
+             transformer
+             (cl-tf:make-transform-stamped
+              base-frame link time
+              (cl-transforms:translation transform)
+              (cl-transforms:rotation transform))
+             :suppress-callbacks t))))
 
-    ;; execute the TF update callbacks
-    (cl-tf:execute-changed-callbacks transformer))))
+      ;; tell the tf transformer about the virtual fixed joints
+      (let* ((bullet-links (btr:link-names ?robot-instance))
+             (urdf-links (cl-urdf:links (btr:urdf ?robot-instance)))
+             ;; get the names of the links, which are only present in the urdf
+             (urdf-link-names (remove-if (lambda (x) (member x bullet-links))
+                                         (loop for name being the hash-keys in urdf-links
+                                               collecting name))))
+        (let ((prev-link-count))
```

Comment:
The second bracket is not necessary here, it is indeed a bit confusing. It should either be something like: 
```lisp
(let ((prev-link-count 0)) ...)
```
or
```lisp
(let (prev-link-count) ...)
```